tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,247 @@
+ * 
+ * affected tables:
+ * - participant_status
+ * - consent_info_history
+ * - consent
+ * - candidate_consent_rel
+ * - candidate_consent_history
+ * 
+ * PHP Version 7
+ * 
+ * @category Main 
+ * @package  Loris
+ * @author   Zaliqa Rosli <zaliqa.rosli@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris
+ */
+
+require_once 'generic_includes.php';
+
+$config = NDB_Config::singleton();
+$db     =& Database::singleton();
+
+// Get consent details from Config.xml
+$consentConfig = $config->getSettingFromXML('ConsentModule');
+$useConsent = $consentConfig['useConsent'];
+$consents = $consentConfig['Consent'];
+
+///////////////////////////////////////
+// VALIDATE EXISTING CONSENT DETAILS //
+///////////////////////////////////////
+
+$errors = [];
+// Check for when consent columns exist in participant_status but not in Config.xml
+$columnQuery = 'SELECT Column_name FROM Information_schema.columns WHERE
+                Column_name LIKE "%consent%" AND Table_name LIKE "participant_status"';
+$existingColumns = $db->pselect($columnQuery, array());
+
+// Format result to only keep consent name columns
+$formattedColumns = [];
+foreach ($existingColumns as $column) {
+    $columnName = $column['Column_name']; 
+    if (!(preg_match("/date/", $columnName) || preg_match("/withdrawal/", $columnName))) {
+        array_push($formattedColumns, $columnName);
+    }
+}
+// Check that column names are in list of consent names
+foreach ($formattedColumns as $columnName) {
+    $i=0;
+    foreach ($consents as $consent){
+        $consentName = $consent['name'];
+        if($consentName===$columnName) {
+            $i++;
+        }
+    }
+    if ($i===0) {
+        array_push(
+            $errors, "The consent type " . $columnName . " exists in the database but not in Config.xml.
+           Please add the consent to Config.xml or delete columns and data from 'participant_status'"
+        );
+    }
+}
+// Check that consent status, date, and withdrawal columns exist in participant_status
+foreach ($consents as $consent) {
+    $consentName       = $consent['name'];
+    $consentDate       = $consentName . "_date";
+    $consentWithdrawal = $consentName . "_withdrawal";
+    $statusExists      = $db->columnExists('participant_status', $consentName);
+    $dateExists        = $db->columnExists('participant_status', $consentDate);
+    $withdrawalExists  = $db->columnExists('participant_status', $consentWithdrawal);
+    if (!($statusExists && $dateExists && $withdrawalExists)) {
+        array_push($errors, "At least one column is missing for " . $consentName . " in participant_status table.
+           Check that status, date, and withdrawal columns exist for this consent type.");
+    } else {
+        // Check for zero dates
+        $dateQuery = 'SELECT ID, CandID, ' . $consentDate . ', ' . $consentWithdrawal . ' FROM participant_status 
```

Comment:
These variables should be database quoted (there's a helper in the DB class to do it for column names)

Reply:
hey @driusan i can't seem to be able to find the function you're talking about .. what is it called?